src/sentry/data_export/tasks.py

```diff
@@ -91,31 +90,61 @@ def process_issues_by_tag(data_export, file, limit, environment_id):
         logger.error("dataexport.error: {}".format(six.text_type(error)))
         raise error
 
-    # Iterate through all the GroupTagValues
     writer = create_writer(file, processor.header_fields)
     iteration = 0
     with snuba_error_handler(logger=logger):
         while True:
             offset = SNUBA_MAX_RESULTS * iteration
             next_offset = SNUBA_MAX_RESULTS * (iteration + 1)
-            gtv_list = processor.get_serialized_data(offset=offset)
-            if len(gtv_list) == 0:
+            gtv_list_unicode = processor.get_serialized_data(offset=offset)
+            if len(gtv_list_unicode) == 0:
                 break
+            # TODO(python3): Remove next line once the 'csv' module has been updated to Python 3
+            # See associated comment in './utils.py'
+            gtv_list = convert_to_ascii(gtv_list_unicode)
             if limit and limit < next_offset:
-                # Since the next offset will pass the limit, write the remainder and quit
                 writer.writerows(gtv_list[: limit % SNUBA_MAX_RESULTS])
                 break
             else:
                 writer.writerows(gtv_list)
                 iteration += 1
 
 
-def process_discover(data_export, file):
-    # TODO(Leander): Implement processing for Discover
-    raise NotImplementedError("Discover processing has not been implemented yet")
+def process_discover(data_export, file, limit, environment_id):
+    """
+    Convert the discovery query to a CSV, writing it to the provided file.
+    """
+    try:
+        processor = DiscoverProcessor(
+            discover_query=data_export.query_info, organization_id=data_export.organization_id
+        )
+    except ExportError as error:
+        metrics.incr("dataexport.error", instance=six.text_type(error))
+        logger.error("dataexport.error: {}".format(six.text_type(error)))
+        raise error
+
+    writer = create_writer(file, processor.header_fields)
+    iteration = 0
+    with snuba_error_handler(logger=logger):
+        while True:
+            offset = SNUBA_MAX_RESULTS * iteration
+            next_offset = SNUBA_MAX_RESULTS * (iteration + 1)
+            raw_data_unicode = processor.data_fn(offset=offset, limit=SNUBA_MAX_RESULTS)["data"]
+            if len(raw_data_unicode) == 0:
+                break
+            # TODO(python3): Remove next line once the 'csv' module has been updated to Python 3
+            # See associated comment in './utils.py'
+            raw_data = convert_to_ascii(raw_data_unicode)
+            raw_data = processor.alias_fields(raw_data)
+            if limit and limit < next_offset:
+                writer.writerows(raw_data[: limit % SNUBA_MAX_RESULTS])
+                break
+            else:
+                writer.writerows(raw_data)
+                iteration += 1
 
 
 def create_writer(file, fields):
-    writer = csv.DictWriter(file, fields)
+    writer = csv.DictWriter(file, fields, extrasaction="ignore")
```

Comment:
This will make sure the discover fields in the browser match the export (no extra `user.id` or `user.username` fields)